applications/GeoMechanicsApplication/custom_elements/interface_element.cpp

```
@@ -187,23 +188,32 @@ void InterfaceElement::Initialize(const ProcessInfo& rCurrentProcessInfo)
-            KRATOS_ERROR_IF_NOT(nodal_stresses.size() == node_ids_common_with_element.size())
-                << " vectors have different sizes" << std::endl;
-        }
-    }
-
     mConstitutiveLaws.clear();
-    for (const auto& r_shape_function_values : shape_function_values_at_integration_points) {
+    for (auto i = std::size_t{0}; i < mIntegrationScheme->GetNumberOfIntegrationPoints(); ++i) {
         mConstitutiveLaws.push_back(GetProperties()[CONSTITUTIVE_LAW]->Clone());
-        mConstitutiveLaws.back()->InitializeMaterial(GetProperties(), GetGeometry(), r_shape_function_values);
+    }
+    // Only interpolate when neighbouring elements that provide nodal stresses were found
+    if (this->Has(NEIGHBOUR_ELEMENTS)) {
+        // interface element can at maximum have 2 neighbours
+        KRATOS_DEBUG_ERROR_IF(this->GetValue(NEIGHBOUR_ELEMENTS).size() > 2)
+            << "Too many neighbour elements for interface element " << this->Id() << std::endl;
+        const auto interface_node_ids = GeometryUtilities::GetNodeIdsFromGeometry(GetGeometry());
+        std::vector<std::optional<Vector>> interface_nodal_cauchy_stresses(interface_node_ids.size());
+        for (auto& r_neighbour_element : this->GetValue(NEIGHBOUR_ELEMENTS)) {
+            std::vector<Vector> neighbour_cauchy_stresses;
+            r_neighbour_element.CalculateOnIntegrationPoints(
+                CAUCHY_STRESS_VECTOR, neighbour_cauchy_stresses, rCurrentProcessInfo);
+            interface_nodal_cauchy_stresses = ExtrapolationUtilities::CalculateNodalVectors(
+                interface_node_ids, r_neighbour_element.GetGeometry(),
+                r_neighbour_element.GetIntegrationMethod(), neighbour_cauchy_stresses,
+                r_neighbour_element.Id());
+            // values of previous iteration completely overwritten. that should probably change
+        }
+        InterpolateNodalStressesToIntegrationPointTractions(interface_nodal_cauchy_stresses);
```

I don't think we need the loop here. The first neighbor will probably do. However, if there is more than one neighbor, we may want to consider the second neighbor as well, in order to check that we arrive at the same prestress. If the prestress values would not match, we can raise an error. But I think building in such a check is beyond the scope of this PR. Removing the loop is within the scope of this PR (and it would address the remark "values of previous iteration completely overwritten. that should probably change").